### ISSUE TITLE
Spurious test failure "type too big for current architecture"

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#55381]
occurred in https://github.com/rust-lang/rust/pull/53821#issuecomment-433349904

```
[01:14:10] ---- [ui (nll)] ui/issues/issue-15919.rs stdout ----
[01:14:10] 
[01:14:10] error: Error: expected failure status (Some(1)) but received status None.
[01:14:10] status: signal: 6
[01:14:10] command: "/checkout/obj/build/tmp/distcheck/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/obj/build/tmp/distcheck/src/test/ui/issues/issue-15919.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/tmp/distcheck/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-15919.nll/a" "-Zborrowck=migrate" "-Ztwo-phase-borrows" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/tmp/distcheck/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/tmp/distcheck/build/x86_64-unknown-linux-gnu/test/ui/issues/issue-15919.nll/auxiliary" "-A" "unused"
[01:14:10] ------------------------------------------
[01:14:10] 
[01:14:10] ------------------------------------------
[01:14:10] stderr:
[01:14:10] stderr:
[01:14:10] ------------------------------------------
[01:14:10] {"message":"the type `[usize; 18446744073709551615]` is too big for the current architecture","code":null,"level":"error","spans":[],"children":[],"rendered":"error: the type `[usize; 18446744073709551615]` is too big for the current architecture\n\n"}
[01:14:10] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[01:14:10] rustc: cxa_atexit.c:100: __new_exitfn: Assertion `l != NULL' failed.
[01:14:10] ------------------------------------------
[01:14:10] 
[01:14:10] thread '[ui (nll)] ui/issues/issue-15919.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3284:9
```

I've even tried running this test in a loop, but it seems to only occur when my machine is running all tests in parallel.

**Comment by RalfJung** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/53429
